core/src/entry_point/start_item_arm.rs

```diff
@@ -49,14 +49,14 @@ pub unsafe extern "C" fn _start(
         // We need to pass app_start, stacktop and app_heap_break to rust_start.
```

Comment:
The changes in this file are incorrect -- it looks like you tried to replace r7 with r6, but r6 was already in use as temporary storage for app_start.

Reply:
This change is part of the "[WIP]" commit which is going to be removed. To be honest, I don't know yet how to update the assembly code. That's why I vote for a rewrite in Rust.

Reply:
Ok, I have an idea...

Reply:
... that works! :smile: